sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuColumnVector.java

```diff
@@ -136,6 +136,8 @@ private static final DType toRapidsOrNull(DataType type) {
       return DType.FLOAT64;
     } else if (type instanceof ByteType) {
       return DType.INT8;
+    } else if (type instanceof BinaryType) {
+      return DType.LIST;
```

Comment:
But it is not just a list.  This is a list of bytes and this API does not have a way to express that.

Reply:
If the code does not need it yes we should remove it.  Otherwise we need to refactor things properly to be able to actually express a proper mapping.

Reply:
Tried removing it, but causes:
```
- Test cast from long *** FAILED ***
  org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 1897.0 failed 1 times, most recent failure: Lost task 0.0 in stage 1897.0 (TID 1907, 10.136.8.146, executor driver): java.l
ang.IllegalArgumentException: BinaryType is not supported for GPU processing yet.
        at com.nvidia.spark.rapids.GpuColumnVector.getRapidsType(GpuColumnVector.java:169)
        at com.nvidia.spark.rapids.GpuCast.doColumnar(GpuCast.scala:250)
        at com.nvidia.spark.rapids.GpuUnaryExpression.columnarEval(GpuExpressions.scala:125)
        at com.nvidia.spark.rapids.RapidsPluginImplicits$ReallyAGpuExpression.columnarEval(implicits.scala:35)
        at com.nvidia.spark.rapids.GpuAlias.columnarEval(namedExpressions.scala:92)
        at com.nvidia.spark.rapids.RapidsPluginImplicits$ReallyAGpuExpression.columnarEval(implicits.scala:35)
        at com.nvidia.spark.rapids.GpuProjectExec$.$anonfun$project$1(basicPhysicalOperators.scala:50)
        at com.nvidia.spark.rapids.RapidsPluginImplicits$MapsSafely.$anonfun$safeMap$1(implicits.scala:160)
        at com.nvidia.spark.rapids.RapidsPluginImplicits$MapsSafely.$anonfun$safeMap$1$adapted(implicits.scala:157)
        at scala.collection.immutable.Stream.foreach(Stream.scala:533)
        at com.nvidia.spark.rapids.RapidsPluginImplicits$MapsSafely.safeMap(implicits.scala:157)
        at com.nvidia.spark.rapids.RapidsPluginImplicits$AutoCloseableProducingSeq.safeMap(implicits.scala:192)
        at com.nvidia.spark.rapids.GpuProjectExec$.project(basicPhysicalOperators.scala:49)
        at com.nvidia.spark.rapids.GpuProjectExec$.projectAndClose(basicPhysicalOperators.scala:40)
        at com.nvidia.spark.rapids.GpuProjectExec.$anonfun$doExecuteColumnar$1(basicPhysicalOperators.scala:89)
        at scala.collection.Iterator$$anon$10.next(Iterator.scala:459)
        at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.gpucolumnartorow_nextBatch_0$(Unknown Source)
        at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
        at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
        at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:729)
        at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:340)
        at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:872)
        at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:872)
        at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
        at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
        at org.apache.spark.rdd.RDD.iterator(RDD.scala:313)
        at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
        at org.apache.spark.scheduler.Task.run(Task.scala:127)
        at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:444)
        at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1377)
        at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:447)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)

Driver stacktrace:
  at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2023)
  at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:1972)
  at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:1971)
  at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
  at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
  at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
  at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:1971)
  at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:950)
  at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:950)
  at scala.Option.foreach(Option.scala:274)
  ...
  Cause: java.lang.IllegalArgumentException: BinaryType is not supported for GPU processing yet.
  at com.nvidia.spark.rapids.GpuColumnVector.getRapidsType(GpuColumnVector.java:169)
  at com.nvidia.spark.rapids.GpuCast.doColumnar(GpuCast.scala:250)
  at com.nvidia.spark.rapids.GpuUnaryExpression.columnarEval(GpuExpressions.scala:125)
  at com.nvidia.spark.rapids.RapidsPluginImplicits$ReallyAGpuExpression.columnarEval(implicits.scala:35)
  at com.nvidia.spark.rapids.GpuAlias.columnarEval(namedExpressions.scala:92)
  at com.nvidia.spark.rapids.RapidsPluginImplicits$ReallyAGpuExpression.columnarEval(implicits.scala:35)
  at com.nvidia.spark.rapids.GpuProjectExec$.$anonfun$project$1(basicPhysicalOperators.scala:50)
  at com.nvidia.spark.rapids.RapidsPluginImplicits$MapsSafely.$anonfun$safeMap$1(implicits.scala:160)
  at com.nvidia.spark.rapids.RapidsPluginImplicits$MapsSafely.$anonfun$safeMap$1$adapted(implicits.scala:157)
  at scala.collection.immutable.Stream.foreach(Stream.scala:533)
  ...

```

Reply:
From looking at the code it looks like it is related to 

https://github.com/NVIDIA/spark-rapids/blob/f3bb5067f32bbf5c6e1ed4d6005e1374d551f7ae/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala#L248

It looks like we are getting the type but not using it in all cases.  For now we should be able to make it lazy, and then be careful not to use it when the type is binary.